jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/GlobalQuartzDBTimerServiceTest.java

```diff
@@ -409,4 +412,81 @@ public void beforeProcessStarted(ProcessStartedEvent event) {
         assertEquals(5, timerExporations.size());
     }
 
+    @Test(timeout = 20000)
+    public void testQuartzJobDeletionOnManagerCloseWithTimerStart() throws Exception {
+        NodeLeftCountDownProcessEventListener countDownListener = new NodeLeftCountDownProcessEventListener("StartProcess", 3);
+        QuartzSchedulerService additionalCopy = new QuartzSchedulerService();
+        additionalCopy.initScheduler(null);
+        // prepare listener to assert results
+        final List<Long> timerExporations = new ArrayList<Long>();
+        ProcessEventListener listener = new DefaultProcessEventListener() {
+
+            @Override
+            public void beforeProcessStarted(ProcessStartedEvent event) {
+                timerExporations.add(event.getProcessInstance().getId());
+            }
+
+        };
+
+        environment = RuntimeEnvironmentBuilder.Factory.get()
+                                                       .newDefaultBuilder()
+                                                       .entityManagerFactory(emf)
+                                                       .addAsset(ResourceFactory.newClassPathResource("org/jbpm/test/functional/timer/TimerStart2.bpmn2"), ResourceType.BPMN2)
+                                                       .schedulerService(globalScheduler)
+                                                       .registerableItemsFactory(new TestRegisterableItemsFactory(listener, countDownListener))
+                                                       .get();
+
+        manager = getManager(environment, false);
+        RuntimeEngine runtime = manager.getRuntimeEngine(ProcessInstanceIdContext.get());
+        KieSession ksession = runtime.getKieSession();
+
+        assertEquals(0, timerExporations.size());
+
+        countDownListener.waitTillCompleted();
+        manager.disposeRuntimeEngine(runtime);
+        int atDispose = timerExporations.size();
+        assertTrue(atDispose > 0);
+
+        ((AbstractRuntimeManager) manager).close(true);
+        countDownListener.reset(1);
+        countDownListener.waitTillCompleted(3000);
+        assertEquals(atDispose, timerExporations.size());
+
+        ksession.destroy();
```

Comment:
@elguardian , regarding the recent error, https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/jbpm-pullrequests/195/testReport/org.jbpm.test.functional.timer/GlobalQuartzDBTimerServiceTest/testQuartzJobDeletionOnManagerCloseWithTimerStart_1_/

We don't need ksession.destroy() here because it's already destroyed by manager.disposeRuntimeEngine()? (in case of PerRequest)